test/distributed/_tensor/test_tensor_ops.py

```diff
@@ -321,21 +321,29 @@ def _test_op(self, mesh, op_call, *args, **kwargs):
     @with_comms
     def test_new_full(self):
         device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        comm_mode = CommDebugMode()
+
+        global_tensor = torch.randn(12, 8)
         placements = [[Shard(0)], [Replicate()]]
         for placement in placements:
-            global_tensor = torch.randn(12, 8)
             input_dt = distribute_tensor(global_tensor, device_mesh, placement)
-            # TODO: Currently CommDebugMode cannot capture communications within DTensor op dispatcher.
-            # Need to revisit the assertions once we have correct CommDebugMode support.
-            comm_mode = CommDebugMode()
             with comm_mode:
-                new_full_dt = input_dt.new_full((4, 8), 42.0)
-                # new_full creates a replicated tensor, which should not trigger any communication.
+                new_full_diff_dt = input_dt.new_full((4, 8), 42.0)
+                # new_full_diff_dt creates a replicated tensor, regardless of input_dt placement,
+                # which should not trigger any communication.
+                self.assertEqual(comm_mode.get_total_counts(), 0)
+            new_full_diff_expected = torch.full((4, 8), 42.0)
+            self.assertTrue(new_full_diff_dt.placements[0].is_replicate())
+            self.assertEqual(new_full_diff_expected, new_full_diff_dt.to_local())
+
+            with comm_mode:
+                new_full_same_dt = input_dt.new_full((12, 8), 42.0)
+                # new_full_same_dt creates a tensor with the same placement as input_dt,
```

Comment:
what is the difference between `new_full_diff_dt` and `new_full_same_dt` test?

Reply:
The difference is in shapes. The former has output tensor with a different shape than the input tensor; the latter's output tensor has the same shape as input.